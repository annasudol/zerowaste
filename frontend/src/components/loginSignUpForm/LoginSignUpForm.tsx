import React, { FC, ReactElement } from 'react';
import { Form, Input, Button } from 'antd';
import { Store } from 'antd/lib/form/interface';

interface AuthProps {
  errorMessage?: string
  handleSubmit(inputValues: Store): void
  signInPage?: boolean
}

const layout = {
  labelCol: { span: 9 },
  wrapperCol: { span: 16 },
};
const tailLayout = {
  wrapperCol: { offset: 8, span: 16 },
};

export const LoginSignUpForm: FC<AuthProps> = ({ errorMessage, handleSubmit, signInPage = false }): ReactElement => {
  const onFinish = (inputValues: Store) => handleSubmit(inputValues);
  return (
    <div className='content overflow-hidden flex justify-center items-center'>
      <div className='flex flex-col'>
        <h1 className='form-header font-bebas uppercase text-darkGray text-center pb-0 m-0'>{!signInPage ? 'Log in' : 'Sign Up'}</h1>
        {errorMessage && <p className='text-coral text-sm text-center mb-3'>{errorMessage}</p>}
        <Form
          {...layout}
          name='basic'
          initialValues={{ remember: true }}
          onFinish={onFinish}
        >
          {signInPage && (
            <Form.Item
              label='Username'
              name='name'
              rules={[{ required: true, type: 'string', min: 5, message: 'Please input your username!' }]}>
              <Input data-testid="name-input"/>
            </Form.Item>
          )}
          <Form.Item
            label='Email'

            name='email'
            rules={[{ required: true, type: 'email', message: 'Please input your email!' }]}
          >
            <Input data-testid="email-input"/>
          </Form.Item>
          <Form.Item
            label='Password'
            name='password'
            rules={[{ required: true, message: 'Please input your password!' }]}
          >
            <Input.Password data-testid="password-input"/>
          </Form.Item>
          <Form.Item {...tailLayout}>
            <Button type='primary' htmlType='submit'>
              Submit
            </Button>
          </Form.Item>
        </Form>
      </div>
    </div>)
}
