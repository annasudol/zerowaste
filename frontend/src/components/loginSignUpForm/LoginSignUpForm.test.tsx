import { shallow } from 'enzyme';
import React from 'react';
import { LoginSignUpForm } from './LoginSignUpForm';
import toJson from 'enzyme-to-json';
import { Button, Form } from 'antd';
import { Store } from 'antd/lib/form/interface';
describe('component', (): void => {
  describe('LoginSignUpForm', (): void => {
    it('match snapshot', (): void => {
      const wrapper = shallow(<LoginSignUpForm handleSubmit={jest.fn()}/>);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('match snapshot with prop signInPage true', (): void => {
        const wrapper = shallow(<LoginSignUpForm handleSubmit={jest.fn()} signInPage/>);

        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('displays error message', (): void => {
        const wrapper = shallow(<LoginSignUpForm handleSubmit={jest.fn()} errorMessage="test"/>);

        expect(wrapper.find('p').text()).toBe("test");
      });
      it('calls handleSubmit upon clicking submit button', (): void => {
        const mockHandleSubmit = jest.fn();
        const wrapper = shallow(<LoginSignUpForm handleSubmit={mockHandleSubmit} />);
        wrapper.find(Form).prop<Function>('onFinish')({email: "example@email.com", password: "Pas321!"});
        wrapper.find(Button).simulate('submit', {email: "example@email.com", password: "Pas321!"}); 
        wrapper.find(Button).at(0).simulate('click');

        expect(mockHandleSubmit).toHaveBeenCalledTimes(1);
      });
});


});