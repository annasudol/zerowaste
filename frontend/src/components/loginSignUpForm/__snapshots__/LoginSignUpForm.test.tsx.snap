// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component LoginSignUpForm match snapshot 1`] = `
<div
  className="content overflow-hidden flex justify-center items-center"
>
  <div
    className="flex flex-col"
  >
    <h1
      className="form-header font-bebas uppercase text-darkGray text-center pb-0 m-0"
    >
      Log in
    </h1>
    <ForwardRef(InternalForm)
      initialValues={
        Object {
          "remember": true,
        }
      }
      labelCol={
        Object {
          "span": 9,
        }
      }
      name="basic"
      onFinish={[Function]}
      wrapperCol={
        Object {
          "span": 16,
        }
      }
    >
      <FormItem
        label="Email"
        name="email"
        rules={
          Array [
            Object {
              "message": "Please input your email!",
              "required": true,
              "type": "email",
            },
          ]
        }
      >
        <Input
          data-testid="email-input"
          type="text"
        />
      </FormItem>
      <FormItem
        label="Password"
        name="password"
        rules={
          Array [
            Object {
              "message": "Please input your password!",
              "required": true,
            },
          ]
        }
      >
        <Password
          action="click"
          data-testid="password-input"
          iconRender={[Function]}
          visibilityToggle={true}
        />
      </FormItem>
      <FormItem
        wrapperCol={
          Object {
            "offset": 8,
            "span": 16,
          }
        }
      >
        <Button
          block={false}
          ghost={false}
          htmlType="submit"
          loading={false}
          type="primary"
        >
          Submit
        </Button>
      </FormItem>
    </ForwardRef(InternalForm)>
  </div>
</div>
`;

exports[`component LoginSignUpForm match snapshot with prop signInPage true 1`] = `
<div
  className="content overflow-hidden flex justify-center items-center"
>
  <div
    className="flex flex-col"
  >
    <h1
      className="form-header font-bebas uppercase text-darkGray text-center pb-0 m-0"
    >
      Sign Up
    </h1>
    <ForwardRef(InternalForm)
      initialValues={
        Object {
          "remember": true,
        }
      }
      labelCol={
        Object {
          "span": 9,
        }
      }
      name="basic"
      onFinish={[Function]}
      wrapperCol={
        Object {
          "span": 16,
        }
      }
    >
      <FormItem
        label="Username"
        name="name"
        rules={
          Array [
            Object {
              "message": "Please input your username!",
              "min": 5,
              "required": true,
              "type": "string",
            },
          ]
        }
      >
        <Input
          data-testid="name-input"
          type="text"
        />
      </FormItem>
      <FormItem
        label="Email"
        name="email"
        rules={
          Array [
            Object {
              "message": "Please input your email!",
              "required": true,
              "type": "email",
            },
          ]
        }
      >
        <Input
          data-testid="email-input"
          type="text"
        />
      </FormItem>
      <FormItem
        label="Password"
        name="password"
        rules={
          Array [
            Object {
              "message": "Please input your password!",
              "required": true,
            },
          ]
        }
      >
        <Password
          action="click"
          data-testid="password-input"
          iconRender={[Function]}
          visibilityToggle={true}
        />
      </FormItem>
      <FormItem
        wrapperCol={
          Object {
            "offset": 8,
            "span": 16,
          }
        }
      >
        <Button
          block={false}
          ghost={false}
          htmlType="submit"
          loading={false}
          type="primary"
        >
          Submit
        </Button>
      </FormItem>
    </ForwardRef(InternalForm)>
  </div>
</div>
`;
